woocommerce/woocommerce-android
# [Hack Week] Allow fixing cha-ching issue from order list

Part of: #10327


### Description
This PR is the second step to implement the soltuon of #10327, this adds logic that allows the user to re-created the orders notification channel, and thus fixing the Cha-ching sound issue from the Order list screen.

**Important:** It would be helpful to review both e92c71598910b050220104926aa2ac4f1cda7da4 and df3a0b75f1d62f9716a970fffbdb7461efb2573c separately, they both just add more properties to some common Event classes that I needed in the PR.

### Testing instructions
1. Open the app information screen on your device, then go to the Notifications menu, then update the "New Order" category to make it silent.
2. Open the app, then open orders list.
3. Notice the dialog.
4. Click on "turn on sound"
5. Click on the "Test sound" button of the snackbar.
6. Confirm a test notification is posted, and uses the Cha-ching sound.
7. Go back to the app information screen, and check that the "New Order" category has been re-created and has the correct sound now.
8. Optional: test a push notification, and confirm it uses the Cha-ching sound.
9. Repeat steps 1 and 2.
10. Click on "Keep silent"
11. Close and re-open the app, then open orders list, and confirm the dialos is not shown.

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/1657201/46c7d73a-a19a-4756-ae5e-fa0758f7418d
